bundler/lib/bundler/settings.rb

```diff
@@ -536,5 +537,34 @@ def self.normalize_uri(uri)
       end
       "#{prefix}#{uri}#{suffix}"
     end
+
+    # This is a hot method, so avoid respond_to? checks on every invocation
+    if :read.respond_to?(:name)
+      def self.key_to_s(key)
+        case key
+        when String
+          key
+        when Symbol
+          key.name
```

Comment:
Oh interesting! So this does not end up allocating any extra strings.

Reply:
yup!